ui-admin/src/portal/PortalEnvConfigView.tsx

```diff
@@ -1,41 +1,92 @@
+    // eslint-disable-next-line
+    // @ts-ignore
+    newConfig[propName] = value
+    setConfig(newConfig)
+  }
+  // saves any changes to the server
+  const save = async (e: React.MouseEvent) => {
+    e.preventDefault()
+    if (!user.superuser) {
+      return
+    }
+    try {
+      const updatedConfig = await Api.updatePortalEnvConfig(portal.shortcode, portalEnv.environmentName, config)
+      Store.addNotification(successNotification('Config saved'))
+      const updatedPortal = _cloneDeep(portal)
+      const matchedEnv = updatedPortal.portalEnvironments
+        .find(updatedPortalEnv => updatedPortalEnv.environmentName === portalEnv.environmentName) as PortalEnvironment
+      matchedEnv.portalEnvironmentConfig = updatedConfig
+      updatePortal(updatedPortal)
+      setConfig(updatedConfig)
+    } catch (e) {
+      Store.addNotification(failureNotification(`Save failed ${  e}`))
+    }
+  }
   return <div className="container p-4">
     <h2 className="h5">Configuration settings - {portalEnv.environmentName}</h2>
     <form className="bg-white p-3">
       <div>
         <label className="form-label">
-        password protected <input type="checkbox" readOnly checked={config.passwordProtected}/>
+        password protected <input type="checkbox" checked={config.passwordProtected}
+            onChange={e => updateConfig('passwordProtected', e.target.checked)}/>
         </label>
       </div>
       <div>
         <label className="form-label">
-        password <input type="text" className="form-control" readOnly={true} value={config.password}/>
+        password <input type="text" className="form-control" value={config.password}
+            onChange={e => updateConfig('password', e.target.value)}/>
         </label>
       </div>
       <div>
         <label className="form-label">
-        accepting registration <input type="checkbox" readOnly checked={config.acceptingRegistration}/>
+        accepting registration
+          <input type="checkbox" checked={config.acceptingRegistration}
+            onChange={e => updateConfig('acceptingRegistration', e.target.checked)}/>
         </label>
       </div>
       <div>
         <label className="form-label">
         participant hostname
-          <input type="text" className="form-control" readOnly={true} value={config.participantHostname}/>
+          <input type="text" className="form-control" value={config.participantHostname}
+            onChange={e => updateConfig('participantHostname', e.target.value)}/>
         </label>
       </div>
       <div>
         <label className="form-label">
         Email source address
-          <input type="text" className="form-control" readOnly={true} value={config.emailSourceAddress}/>
+          <input type="text" className="form-control" value={config.emailSourceAddress}
+            onChange={e => updateConfig('emailSourceAddress', e.target.value)}/>
         </label>
       </div>
+      <button onClick={save}
```

Comment:
Since #439 has merged, the Button component added there may be useful here. It handles aria-disabled, disabled styles, and tooltips.